test-data/unit/check-classes.test

```diff
@@ -4009,14 +4010,25 @@ class B(A):
 main:4: error: Incompatible types in assignment (expression has type "str", base class "A" defined the type as "int")
 main:5: error: Incompatible types in assignment (expression has type "str", base class "A" defined the type as "int")
 
-[case testClassIgnoreType]
+[case testClassIgnoreType_RedefinedAttributeAndGrandparentAttributeTypesNotIgnored]
 class A:
     x = 0
 class B(A):
     x = ''  # type: ignore
 class C(B):
     x = ''
 [out]
+main:6: error: Incompatible types in assignment (expression has type "str", base class "A" defined the type as "int")
```

Comment:
Same as above. A user might want to ignore the error _once_ in the base class, instead of ignoring it in _every_ subclass.

Reply:
This is slightly different than the above situation, as here, there was no error before.

I think it depends on how you want to interpret the `# type: ignore`.
In the current behavior, because `B.x` has an ignore, you are no longer warned when you redefine `x` in `C.x` that that redefinition is actually incompatible with ancestor's `A.x`.

A library author may have defined `B.x` with the ignore comment, and let us say that you are writing class `C` yourself. I would find it unexpected that because the library author chose to add an ignore comment on `B.x` that that would suppress a certain type of error when defining my own class.
I may not know in detail how `B` is defined/type-annotated, so I may reasonably expect that the type checker would perform its normal checks for my definition of class `C` (in this case, that it is compatible with `A`).

Are there other places in mypy where `# type: ignore` comments "spread" in that sense?

Reply:
Oops:
~(in this case, that it is compatible with `A`)~
should be:
(in this case, that `C.x` is actually compatible with `A.x`)